src/nccl_ofi_net.c

```diff
@@ -768,77 +771,78 @@ static int find_ofi_provider(struct fi_info **providers)
 
 /*
  * @brief	Calls fi_getinfo() to find a list of usable providers for RDM
- *		tagged endpoints.
+ *		tagged endpoints.  The code will return all providers
+ *		with the same name as the first provider in the list
+ *		returned by fi_getinfo() that satisfies any filters
+ *		applied by prov_include.
  *
- * @param	prov_include_list
+ * @param	prov_include
  *		Contains a list of preferred provider names.
  *
  * @return	A list of fi_info structures for a single provider.
  * @return	0 on success
  *		non-zero on error
  */
-static int get_ofi_provider(char *prov_include, struct fi_info **prov_info_list)
+static int get_ofi_provider(const char *prov_include, struct fi_info **prov_info_list)
```

Comment:
Is there any possibility that get_ofi_provider() would be called concurrently with another routine that used strtok()?  (which is used in the in_list() utility routine, invoked here)

Reply:
no.  `init()` is serialized.